### ISSUE TITLE
Unable to use .Take on navigation property.

### ISSUE THREAD

**Issue by mcchessers-sp** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1782]
### Steps to reproduce
Hi. I have an aggregate root "Player" with a collection "ActiveDays" via a "PlayerId" foreign key. When I try to apply "Take()" to the query to limit the number of results in that collection, like so:

`DbContext.Players.Where(p => p.Id == (int)TestPlayers[0].Id).Include(p => p.ActiveDays.Take(1)).FirstOrDefault();
`

I get the error 

`System.InvalidOperationException : The LINQ expression 'ROW_NUMBER() OVER(PARTITION BY u0.user_id ORDER BY u0.user_id ASC, u0.day_active ASC)' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to 'AsEnumerable', 'AsAsyncEnumerable', 'ToList', or 'ToListAsync'. See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.`

(Note that I'm mapping the column "user_id" to "PlayerId" as this is a legacy schema that has some inconsistencies. The ActiveDays table has a composite primary key of user_id and day_active.)

Attempting to add a "AsEnumerable" or similar results in
System.InvalidOperationException : The expression 'player.ActiveDays.AsEnumerable().AsQueryable().Take(__limit_1)' is invalid inside an 'Include' operation, since it does not represent a property access: 't => t.MyProperty'. To target navigations declared on derived types, use casting ('t => ((Derived)t).MyProperty') or the 'as' operator ('t => (t as Derived).MyProperty'). Collection navigation access can be filtered by composing Where, OrderBy(Descending), ThenBy(Descending), Skip or Take operations. For more information on including related data, see http://go.microsoft.com/fwlink/?LinkID=746393.

Is there something I am misunderstanding or is this an issue?

### Further technical details

MySQL version: 5.7.22
Operating system: Windows 10
Pomelo.EntityFrameworkCore.MySql version: 7.0.0
Microsoft.AspNetCore.App version: 7.0

**Comment by austinpapritz** [comment]
I was able to reproduce your error. Based on Microsoft [docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.entityframeworkqueryableextensions.include?view=efcore-5.0) for `.Include()`, it expects a navigation property related to the entity being queried. Thus, it sees `p.ActiveDays.Take(1)` as the name of the navigation property you are trying to include in the query, it will not perform `.Take()` on the property inside of an `.Include()` as you are expecting. 

You can perform the `.Take(1)` right outside it like this `.Include().Take(1)`. This will limit the entire query to 1 to one Player, which I'm not sure what you are trying to do. Otherwise, you can limit the `ActiveDays` property afterwards like this
`player.ActiveDays = player.ActiveDays.Take(1).ToList();`

**Comment by austinpapritz** [comment]
I was able to reproduce your error with MySQL version 5.7.22.

![mysql5722](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/assets/109837655/35c719ad-f785-42ef-b2bb-ca808fe95430)

Then I upgraded to MySQL 8.0.34 and the debugger didn't catch anything there, although Swagger UI threw a 500 because of object cycling (due to my simplified recreation of your project). Either way, I believe this shows it's likely a MySQL 5.722 problem.

![mysql8034](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/assets/109837655/665a8951-d70c-42d7-b8c4-5320c9f5b8b0)
